vshard/storage/init.lua

```diff
@@ -1597,6 +1609,7 @@ local function bucket_recv_xc(bucket_id, from, data, opts)
             return nil, lerror.box(err)
         end
         box.begin()
+        M._on_bucket_event:run(event_type, bucket_id, space_name)
```

Comment:
We should change the last argument to be more extendible. To this format:
```
{
    spaces = {name1, name2, ...}
}
```
Which would be in our case so far always `{spaces = {space_name}}`.
The reasons are:
- There might be multiple spaces per-txn potentially. In fact, I think that is a matter of short time. See https://github.com/tarantool/vshard/issues/346. With foreign keys we will have to touch multiple spaces per txn;
- We can't just provide storage engine name instead of space names, because in the future the transaction manager might become single for both engines. Hence allowing FKs between engines. Thus making txns engine-specific only temporarily;
- When the space names array is wrapped into a dict, it allows us to add here more keys than just `spaces` in the future, if needed.

All the same for bucket GC.

Reply:
Fixed